### PR TITLE
remove array datatype

### PR DESCRIPTION
**Pull request by sleidig**
closes: #2368 

deployed to https://dev.aam-digital.net for testing on an existing database

### CODE REVIEW

File: doc/compodoc_sources/concepts/configuration.md [hunk 1]
```diff
@@ -450,15 +450,15 @@ In the List View columns config, use the "DisplayConfigurableEnum" component for
...
 The following example creates a property `materials` where multiple values from the `materials` configurable enum can be selected.
 
 ```json
 {
   "name": "materials",
   "schema": {
-    "dataType": "array",
-    "innerDataType": "configurable-enum",
+    "dataType": "configurable-enum",
+    "dataArray": true,
```
Comment (sleidig) [inline doc/compodoc_sources/concepts/configuration.md:461]: happy to discuss naming: isArray (clear and simple) or dataArray (same prefix as "dataType", which it relates to) came to my mind

File: src/app/child-dev-project/attendance/model/event-attendance.ts [hunk 2]
```diff
@@ -47,3 +47,14 @@ export class EventAttendance {
     return Object.assign(new EventAttendance(), this);
   }
 }
+
+/**
+ * A full registry of event-attendance entries for multiple participants
```
Comment (sleidig) [inline src/app/child-dev-project/attendance/model/event-attendance.ts:52]:
```suggestion
 * A full registry of event-attendance entries for multiple participants.
 *
 * TODO: this class can become the basis for a more generic attendance data that is not hard-wired to Note entities.
```

File: src/app/core/basic-datatypes/entity-array/display-entity-array/display-entity-array.component.html [hunk 3]
```diff
@@ -1,13 +0,0 @@
-<ng-container *ngIf="entities">
```
Comment (sleidig) [inline src/app/core/basic-datatypes/entity-array/display-entity-array/display-entity-array.component.html:1]: merged with display-entity.component

File: src/app/core/basic-datatypes/entity/entity-block/entity-block.component.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<span
```
Comment (sleidig) [inline src/app/core/basic-datatypes/entity/entity-block/entity-block.component.html:1]: actually renamed from display-entity.component and turned into a simpler component here

File: src/app/core/entity/entity-actions/entity-anonymize.service.ts [hunk 5]
```diff
@@ -77,11 +78,21 @@ export class EntityAnonymizeService extends CascadingEntityAction {
       entity.getSchema().get(key).dataType,
     );
 
-    entity[key] = await dataType.anonymize(
-      entity[key],
-      entity.getSchema().get(key),
-      entity,
-    );
+    let anonymizedValue;
+    if (entity.getSchema().get(key).dataArray) {
```
Comment (tomwwinter) [inline src/app/core/entity/entity-actions/entity-anonymize.service.ts:82]: Can this produce null-pointer errors?

File: src/app/core/import/import.service.ts [hunk 6]
```diff
@@ -164,7 +163,7 @@ export class ImportService {
         if (!!parsed || parsed === 0) {
           // enforcing array values to be correctly assigned
           entity[mapping.propertyName] =
-            isArrayProperty(entityConstructor, mapping.propertyName) &&
+            entityConstructor.schema.get(mapping.propertyName).dataArray &&
```
Comment (tomwwinter) [inline src/app/core/import/import.service.ts:166]: Can this produce null-pointer erros?

File: src/app/core/common-components/entities-table/entities-table.component.ts [hunk 7]
```diff
@@ -339,7 +338,7 @@ export class EntitiesTableComponent<T extends Entity> implements AfterViewInit {
   private disableSortingHeaderForAdvancedFields(c: FormFieldConfig) {
     // if no dataType is defined, these are dynamic, display-only components
     if (
-      c.dataType === EntityArrayDatatype.dataType ||
+      // TODO: prevent sort for array data?
```
Comment (tomwwinter) [inline src/app/core/common-components/entities-table/entities-table.component.ts:341]: Does not make so much sense to sort by a bunch of values. I would prevent the sort here.

File: src/app/core/basic-datatypes/entity/display-entity/display-entity.component.scss [hunk 8]
```diff
@@ -1,13 +1,7 @@
-@use "variables/sizes";
+@use "../../../../../styles/variables/sizes";
```
Comment (tomwwinter) [inline src/app/core/basic-datatypes/entity/display-entity/display-entity.component.scss:1]: Is this intended?

Comment (sleidig) [reply src/app/core/basic-datatypes/entity/display-entity/display-entity.component.scss:1]:
```suggestion
@use "variables/sizes";
```
no, I don't think so ...

File: src/app/core/common-components/entities-table/entities-table.component.ts (hunk 7 above)
Comment (sleidig) [inline src/app/core/common-components/entities-table/entities-table.component.ts:341]:
```suggestion
      c.dataArray ||
```

File: src/app/core/entity/entity-actions/entity-anonymize.service.ts (hunk 5 above)
Comment (sleidig) [reply src/app/core/entity/entity-actions/entity-anonymize.service.ts:82]: this is called from an iteration of all schema fields, so we can safely assume these are defined.

Thanks for flagging this, though. We do have issue on this from time to time ...

**Review by tomwwinter — APPROVED** [review]

**Pull request merged**